app/javascript/configureStore.js

```diff
@@ -1,48 +1,34 @@
 import { createStore, applyMiddleware, combineReducers } from 'redux'
 import loginReducer from './reducers/loginReducer'
 import signupReducer from './reducers/signupReducer'
+import productReducer from './reducers/productReducer'
 import thunk from 'redux-thunk'
 import { composeWithDevTools } from 'redux-devtools-extension'
 import Datastore from './datastore';
 
 const datastore = new Datastore()
 
 export const INITIAL_STATE = {
-  thingsReducer: {
-    things: [{
-      name: 'test',
-      guid: '123'
-    }]
-  },
   loginReducer: {
     login: null,
     loggedin: datastore.get('isLoggedIn'),
-    userdata: datastore.get('userdata'),
+    firstname: datastore.get('firstname'),
     token: datastore.get('token')
   },
   signupReducer: {
     signup: null,
     signuperror: null,
     signedup: false
+  },
+  productReducer: {
+    products: null
```

Comment:
I would default this to an empty array to indicate that there are no products. So that on the other side you don't have to check whether `products` is equal to null, and you can just run `map` or `each` over the array.